compiler/rustc_middle/src/ty/context.rs

```diff
@@ -2251,9 +2251,15 @@ impl<'tcx> TyCtxt<'tcx> {
     /// Find the crate root and the appropriate span where `use` and outer attributes can be
     /// inserted at.
     pub fn crate_level_attribute_injection_span(self, hir_id: HirId) -> Option<Span> {
-        for (_hir_id, node) in self.hir().parent_iter(hir_id) {
-            if let hir::Node::Crate(m) = node {
-                return Some(m.spans.inject_use_span.shrink_to_lo());
+        for (_hir_id, node) in
+            [(hir_id, self.hir_node(hir_id))].into_iter().chain(self.hir().parent_iter(hir_id))
```

Comment:
I'd rather we not fix this in the `parent_iter` caller given that it's a bit specific solution to a general problem.

I think we should fix this in `TyCtxtFeed::feed_hir` -- I'll open a PR, but I'd rather get @petrochenkov to review it as the author of this `Node::Synthetic` stuff who might have opinions on the API.